clients/erigon/Dockerfile.git

```diff
@@ -1,45 +1,54 @@
-# Builds erigon from a git repository.
+## Build Erigon From Git:
+
+# Builder stage: Compiles erigon from a git repository
 FROM golang:1-alpine as builder
+
 ARG user=ledgerwatch
 ARG repo=erigon
 ARG branch=devel
 
-RUN \
-    apk add --update build-base linux-headers git bash jq          \
-    ca-certificates libstdc++ libgcc musl-dev                   && \
-    git clone --depth 1 --branch $branch                           \
-    https://github.com/$user/$repo                              && \
-    (cd erigon && make erigon)                                  && \
-    (cd erigon                                                  && \
-    echo "{}"                                                      \
+RUN apk add --no-cache \
+    build-base \
+    linux-headers \
+    git \
+    bash \
+    jq \
+    ca-certificates
+
+RUN git clone --depth 1 --branch $branch https://github.com/$user/$repo \
+    && cd erigon \
+    && make erigon \
+    && echo "{}" \
     | jq ".+ {\"repo\":\"$(git config --get remote.origin.url)\"}" \
-    | jq ".+ {\"branch\":\"$(git rev-parse --abbrev-ref HEAD)\"}"  \
-    | jq ".+ {\"commit\":\"$(git rev-parse HEAD)\"}"               \
-    > /version.json)
+    | jq ".+ {\"branch\":\"$(git rev-parse --abbrev-ref HEAD)\"}" \
+    | jq ".+ {\"commit\":\"$(git rev-parse HEAD)\"}" > /version.json
 
+# Final stage: Sets up the environment for running erigon
 FROM alpine:latest
-RUN apk add --update bash curl jq libstdc++ libgcc
 
 # Copy compiled binary from builder
-COPY --from=builder /build/bin/* /usr/local/bin/
+COPY --from=builder /go/erigon/build/bin/* /usr/local/bin/
 
-# Create version.txt
-RUN /usr/local/bin/erigon --version | sed -e 's/erigon version \(.*\)/\1/' > /version.txt
+RUN apk add --no-cache \
+    bash \
+    curl \
+    jq \
+    libstdc++ \
+    libgcc
```

Comment:
Why are thes (libstdc++, libgcc) needed in the "not-builder"?

Reply:
Ahh so I believe these are already included in the default Dockerfile base image with a pre-built Erigon. 

As we are building from source here (‘.git’) and also in ‘.local’ : in the “not-builder” stage where we are running the client, these dependencies are needed to run the client successfully